src/steps/02-EvaluationCriteria/MRR/CertificationPOCTypeForm.vue

```diff
@@ -0,0 +1,415 @@
+    const title = this.title;
+    return {
+      sys_id,
+      first_name,
+      last_name,
+      middle_name,
+      role,
+      rank_components,
+      suffix,
+      salutation,
+      phone: phone || "",
+      phone_extension: phoneExt || "", // not used on Mission Owner contact entry form
+      email,
+      type: this.selectedOptionType as string,
+      dodaac: "",
+      can_access_package: "true",
+      grade_civ,
+      title,
+      manually_entered: "", // not used on Mission Owner contact entry form
+    };
+  }
+
+  /**
+   * Checks and returns if the fair opportunity data POC type data has changed.
+   */
+  private hasFairOpportunityDataChanged(): boolean {
+    const currentData = {} as FairOpportunityDTO;
+    currentData[this.POCTypePropName] = this.selectedOptionType;
+    const savedData = {} as FairOpportunityDTO;
+    const savedFairOpportunity = AcquisitionPackage.fairOpportunity as FairOpportunityDTO;
+    savedData[this.POCTypePropName] = savedFairOpportunity[this.POCTypePropName] || "";
+    return hasChanges(currentData, savedData);
+  }
+
+  private hasCurrentContactFormChanged(): boolean {
+    return hasChanges(this.currentContactFormData, this.certificationPOCContactDTO);
+  }
+
+  /**
+   * Checks to see if the certification POC type changed and takes one of
+   * the actions below.
+   * 1. If the user updates the POC type from "NEW" to one of the other 3 types, then
+   *      DELETES the contact associated with "NEW" and updates the fair opportunity
+   *      properties accordingly
+   * 2. If the user changes the POC type from one of the other 3 types to "NEW",
+   *      then creates a NEW contact and updates the fair opportunity
+   *      properties accordingly
+   * 3. If the user changes the POC type among one of the 3 existing contact types,
+   *      then just updates the fair opportunity properties.
+   * 4. If the POC type has not changed, then checks if any of the form data has changed
+   *      and makes a call to save the changed data.
+   */
+  @Watch('_saveForm')
+  protected async saveOnLeave(): Promise<boolean> {
+    try {
+      await AcquisitionPackage.setValidateNow(true);
+      const savedPOCType =
+          (AcquisitionPackage.fairOpportunity as FairOpportunityDTO)[this.POCTypePropName];
+      const fairOpportunity = {} as FairOpportunityDTO;
+      if (this.selectedOptionType === "NEW") {
+        // user changed from other types to NEW or could be first time filling the form.
+        if (this.hasCurrentContactFormChanged()) {
+          let savedContact = await ContactData.saveContact(this.currentContactFormData);
+          savedContact = convertColumnReferencesToValues(savedContact);
+          fairOpportunity[this.POCTypePropName] = this.selectedOptionType;
+          fairOpportunity[this.POCPropName] = savedContact.sys_id;
+          await AcquisitionPackage.setFairOpportunity(fairOpportunity);
+          this.selectedOption.value = savedContact.sys_id as string;
```

Comment:
`this.selectedOption` is a getter. You can't set a value directly like this on a getter.